### PR TITLE
Prevent sixense manager from triggering continuous reset

### PR DESCRIPTION
**Pull request by jherico**

### CODE REVIEW

File: libraries/input-plugins/src/input-plugins/SixenseManager.cpp [hunk 1]
```diff
@@ -221,21 +215,23 @@ static const float MINIMUM_ARM_REACH = 0.3f; // meters
...
-        if (_calibrationState == CALIBRATION_STATE_IDLE) {
-            return;
-        }
+    // Calibration buttons aren't set, so check the state, and request a reset if necessary.
+    if (!calibrationRequested(controllers)) {
         switch (_calibrationState) {
+            case CALIBRATION_STATE_IDLE:
+                return;
             case CALIBRATION_STATE_COMPLETE:
                 {
```
Comment (ZappoMan) [inline libraries/input-plugins/src/input-plugins/SixenseManager.cpp]: note: I realize you didn't write this code, so not a merge blocker, but this switch statement formatting is completely inconsistent with the coding standard.

File: libraries/input-plugins/src/input-plugins/SixenseManager.h [hunk 2]
```diff
@@ -65,9 +79,8 @@ public slots:
         virtual void focusOutEvent() override;
 
         void handleButtonEvent(unsigned int buttons, bool left);
-        void handleAxisEvent(float x, float y, float trigger, bool left);
         void handlePoseEvent(float deltaTime, glm::vec3 position, glm::quat rotation, bool left);
-        void updateCalibration(void* controllers);
+        void updateCalibration(_sixenseControllerData* controllers);
```
Comment (ZappoMan) [inline libraries/input-plugins/src/input-plugins/SixenseManager.h]: Coding Standard: wth?? Is _sixenseControllerData a data type? or a member variable? we shouldn't have a data type with an "_" in it's name.

Comment (ZappoMan) [inline libraries/input-plugins/src/input-plugins/SixenseManager.h]: ugh! I just saw that this is a Sixense data structure name... bleh!

File: libraries/input-plugins/src/input-plugins/SixenseManager.cpp (hunk 1 above)
Comment (jherico) [inline libraries/input-plugins/src/input-plugins/SixenseManager.cpp]: fixed.

File: libraries/input-plugins/src/input-plugins/SixenseManager.h [hunk 3]
```diff
@@ -19,6 +19,20 @@
 
 #include "InputPlugin.h"
 
+class QLibrary;
+
+const unsigned int BUTTON_0 = 1U << 0; // the skinny button between 1 and 2
```
Comment (ZappoMan) [inline libraries/input-plugins/src/input-plugins/SixenseManager.h]: aren't these already defined in sixense.h?

```
#define SIXENSE_BUTTON_BUMPER   (0x01<<7)
#define SIXENSE_BUTTON_JOYSTICK (0x01<<8)
#define SIXENSE_BUTTON_1        (0x01<<5)
#define SIXENSE_BUTTON_2        (0x01<<6)
#define SIXENSE_BUTTON_3        (0x01<<3)
#define SIXENSE_BUTTON_4        (0x01<<4)
#define SIXENSE_BUTTON_START    (0x01<<0)
```

Is there a reason we need them here? Can't we just `#ifdef HAVE_SIXENSE` in places where we need to use these defines?

Comment (jherico) [inline libraries/input-plugins/src/input-plugins/SixenseManager.h]: bad merge artifact, fixed.

File: libraries/input-plugins/src/input-plugins/SixenseManager.h (hunk 2 above)
Comment (jherico) [inline libraries/input-plugins/src/input-plugins/SixenseManager.h]: fixed with a `using` statement.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2ecc4f8a5c88dd32d3d3ba05f8a01c3a119d23f1 available at: https://gist.github.com/014ed6c807887511bcaa<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6349.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6349.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 14b2de167d346d3b1246c142029b039f2b868ffb available at: https://gist.github.com/eae97fe3ecc97eb7d97f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6349.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6349.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 29dda051a8180ee7dc722c6c9fd504008baef344 available at: https://gist.github.com/014ed6c807887511bcaa<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6349.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6349.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 07be03dc7e244ec4dcc27f18fcc1c595deff5a93 available at: https://gist.github.com/014ed6c807887511bcaa<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6349.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6349.dmg

**Pull request merged**